Pathogens/Utility/location_utils/lat_lng_generation_report.py

```diff
@@ -0,0 +1,314 @@
+
+    return prefix + " - " + info_statement + " - " + details + " - " + suggestion
+
+def generate_line_for_city_state_bounding_box_consistency_check(
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    mapbox_response,\
+):
+    if(city_name is None or state_name is None):
+        return
+
+    if(mapbox_response is None):
+        return
+    
+    mapbox_city_info_response = mapbox_api_client.make_mapbox_request(
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name,
+        lat_lng_report_file = None,
+        mapbox_request_param_override = None
+    )
+    mapbox_state_info_response = mapbox_api_client.make_mapbox_request(
+        city_name = None,
+        state_name = state_name,
+        country_name = country_name,
+        lat_lng_report_file = None,
+        mapbox_request_param_override = None
+    )
+
+    if(mapbox_city_info_response is None or mapbox_state_info_response is None):
+        return
+    
+    city_coords = mapbox_city_info_response.center_coordinates
+    state_bounding_box = mapbox_state_info_response.bounding_box
+
+    if(city_coords is None or state_bounding_box is None):
+        return
+
+    if(is_point_in_bounding_box(point = city_coords, bounding_box = state_bounding_box)):
+        return
+
+    region_name_as_string = ('"' + mapbox_response.region_name + '"') if mapbox_response.region_name is not None else "N/A"
+    city_name_as_string = ('"' + city_name + '"') if city_name is not None else "N/A"
+    state_name_as_string = ('"' + state_name + '"') if state_name is not None else "N/A"
+    
+    prefix = generate_mapbox_api_request_log_prefix(
+        logLevel = MapboxApiRequestLogLevel.ERROR,
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name
+    )
+    info_statement = "[City listed and state listed are incompatible with one another]"
+    details = '[City name: ' + city_name_as_string + ', State name: ' + state_name_as_string + ']'
+    suggestion = "[According to mapbox, the state this city belongs to is " + region_name_as_string + ". If there are no other errors and you agree with this state placement, consider moving the city to that state. If that state name sounds wrong, please feel free to omit the city name since it might be resulting in the pin being in the incorrect state. Please also check the other errors to see if they provide any more context here.]"
+
+    return prefix + " - " + info_statement + " - " + details + " - " + suggestion
+
+def generate_line_for_invalid_city_but_valid_state_check(
```

Comment:
Context: Some of our problems come from someone giving a state name as a city name. Mapbox usually returns `None` in instances like that but sometimes it tries to take an educated guess and it's almost always wrong. Ex `Tyrol, Tyrol, Austria` returns `None` so it falls back to `N/A, Tyrol, Austria` which is valid but in cases like `Madhya Pradesh, Madhya Pradesh, India` it legit just guesses and gets it totally wrong so because this is a mixed bag some of these warnings are actual problems and some of them are just annoying

For what it's worth I think examples like `Tyrol, Tyrol, Austria` should be corrected to `N/A, Tyrol, Austria` but let's merge the report first and then I'll go talk to the data team about how they feel about me making my changes to Airtable.